Change error messages to exclude line numbers and file names

Right now:
```
gore> "this"+"is"+"an"+error
# command-line-arguments
/tmp/726836892/gore_session.go:10:47: type error is not an expression
error: exit status 2
exit status 2
```
returns line numbers which make no sense in the context. Cannot everything up to the beginning of the error be suppresed? Also `#command-line-arguments`, which seems to be always the same?

--- a/errfilter.go
+++ b/errfilter.go
@@ -1,0 +1,54 @@
+package main
+
+import (
+	"bytes"
+	"io"
+)
+
+type errFilter struct {
+	w   io.Writer
+	buf []byte
+}
+
+func newErrFilter(w io.Writer) *errFilter {
+	return &errFilter{w, nil}
+}
+
+func (w *errFilter) Write(p []byte) (n int, err error) {
+	var m, i int
+	for {
+		i = bytes.IndexRune(p, '\n')
+		if i < 0 {
+			break
+		}
+		if len(w.buf) > 0 {
+			m, err = w.w.Write(w.replace(w.buf))
+			n += m
+			w.buf = nil
+			if err != nil {
+				return
+			}
+		}
+		m, err = w.w.Write(w.replace(p[:i+1]))
+		n += m
+		if err != nil {
+			return
+		}
+		p = p[i+1:]
+	}
+	w.buf = p
+	return
+}
+
+func (w *errFilter) replace(p []byte) []byte {
+	if bytes.HasPrefix(p, []byte("# command-line-arguments")) {
+		return nil
+	}
+	if i := bytes.Index(p, []byte("gore_session.go")); i >= 0 {
+		if j := bytes.IndexRune(p[i:], ' '); j >= 0 {
+			return p[i+j+1:]
+		}
+		return p
+	}
+	return p
+}

--- a/main.go
+++ b/main.go
@@ -129,8 +129,9 @@ func main() {
 				continue
 			} else if err == ErrQuit {
 				break
+			} else if err != ErrCmdRun {
+				fmt.Println(err)
 			}
-			fmt.Println(err)
 		}
 		rl.Accepted()
 	}
@@ -364,6 +365,7 @@ type Error string
 const (
 	ErrContinue Error = "<continue input>"
 	ErrQuit     Error = "<quit session>"
+	ErrCmdRun   Error = "<command failed>"
 )
 
 func (e Error) Error() string {
@@ -469,7 +471,8 @@ func (s *Session) Eval(in string) error {
 				}
 			}
 		}
-		errorf("%s", err)
+		debugf("%s", err)
+		err = ErrCmdRun
 	}
 
 	return err

--- a/main.go
+++ b/main.go
@@ -288,7 +288,7 @@ func (s *Session) goRun(files []string) error {
 	cmd := exec.Command("go", args...)
 	cmd.Stdin = os.Stdin
 	cmd.Stdout = s.stdout
-	cmd.Stderr = s.stderr
+	cmd.Stderr = newErrFilter(s.stderr)
 	return cmd.Run()
 }
 

--- a/session_test.go
+++ b/session_test.go
@@ -172,3 +172,24 @@ func TestRun_Const(t *testing.T) {
 `, stdout.String())
 	require.Equal(t, "", stderr.String())
 }
+
+func TestRun_Error(t *testing.T) {
+	stdout, stderr := new(bytes.Buffer), new(bytes.Buffer)
+	s, err := NewSession(stdout, stderr)
+	require.NoError(t, err)
+
+	codes := []string{
+		`foo`,
+		`len(100)`,
+	}
+
+	for _, code := range codes {
+		err := s.Eval(code)
+		require.Error(t, err)
+	}
+
+	require.Equal(t, "", stdout.String())
+	require.Equal(t, `undefined: foo
+invalid argument 100 (type int) for len
+`, stderr.String())
+}

